src/emulator/gui/src/gui.cpp

```diff
@@ -230,6 +251,14 @@ void init(HostState &host) {
     init_icons(host);
     if (!host.cfg.background_image.empty())
         init_background(host, host.cfg.background_image);
+
+    HostState *host_pointer = &host;
+
+    // Initialize trophy callback
+    host.np.trophy_state.trophy_unlock_callback = [host_pointer](NpTrophyUnlockCallbackData &callback_data) {
+        const std::lock_guard<std::mutex> guard(host_pointer->gui.trophy_unlock_display_requests_access_mutex);
+        host_pointer->gui.trophy_unlock_display_requests.push(std::move(callback_data));
+    };
```

Comment:
Good point.

Second question: We can't get rid of it (if im right). Passing the host reference as reference (needed!) will be obselete since if the construct function goes out of scope, so do the reference.

Reply:
Okay, that's not how I thought references work (I kinda thought they were just pointers with special syntax)... Fair enough.

Reply:
They are pointer in disguise yes. Though if you pass it as reference or either copy, the capture value will goes out of scope when this construct function is done.